WcaOnRails/app/webpacker/components/Delegates/index.jsx

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React from 'react';
+
+import { Button } from 'semantic-ui-react';
+import I18n from '../../lib/i18n';
+import UserBadge from '../UserBadge';
+
+import '../../stylesheets/delegates/style.scss';
+
+export default function Delegates({
+  delegates,
+  isEditVisible,
+}) {
+  const [seniorDelegates, setSeniorDelegates] = React.useState([]);
+  const [delegatesUnderSeniorDelegates, setDelegatesUnderSeniorDelegates] = React.useState([]);
+
+  // TO_VERIFY: I assume there are no cases where there are no delegates without
+  // senior delegate unless they are senior delegate
+
+  React.useEffect(() => {
+    setSeniorDelegates(delegates
+      .filter((user) => user.delegate_status === 'senior_delegate')
+      .sort((user1, user2) => {
+        if (user1.region < user2.region) {
+          return -1;
+        } if (user1.region > user2.region) {
+          return 1;
+        }
+        return 0;
+      }));
+  }, []);
```

Comment:
This effect is missing dependencies. If it's really just meant to run once initially then you could just pass this function to the `useState` for `seniorDelegates` to be the initial value.
If it is meant to change whenever `delegates` changes then similarly to my comment below, just having a const which calculates this directly, possibly wrapped in a `useMemo`, would be cleaner.

Reply:
Okay, done.